src/receipt-accumulator.ts

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { AccumulatedReceipt } from "./sync-accumulator";
+import { MapWithDefault } from "./utils";
+
+export class ReceiptAccumulator {
+    private readReceipts: Map<string, AccumulatedReceipt> = new Map();
+    private threadedReadReceipts: MapWithDefault<string, Map<string, AccumulatedReceipt>> = new MapWithDefault(
```

Comment:
Please add a doc for the string keys here as well.

Reply:
Done in next PR